calls/Space.php

```diff
@@ -161,4 +169,59 @@ public static function leaveSpace($event)
             );
         }
     }
+
+    /**
+     * Enable ethereum integration for already existing spaces
+     *
+     * @param $event
+     * @throws GuzzleException
+     * @throws HttpException
+     * @throws Exception
+     */
+    public function enable($event)
+    {
+        $space = $event->sender;
+
+        if (!$space instanceof BaseSpace) {
+            return;
+        }
+
+        $spaceDefaultAccount = Account::findOne([
+            'space_id' => $space->id,
+            'account_type' => Account::TYPE_DEFAULT
+        ]);
+
+        if (!$spaceDefaultAccount) {
+            Utils::createDefaultAccount($space);
+        } else {
+            Wallet::createWallet(new Event(['sender' => $spaceDefaultAccount]));
+            Dao::createDao($event);
+        }
+
+        $asset = Utils::issueSpaceAsset($space);
+
+        $transactions = Transaction::findAll([
+            'asset_id' => $asset->id,
+            'transaction_type' => Transaction::TRANSACTION_TYPE_ISSUE
+        ]);
+
+        // mint coins foreach issue transaction of the space
+        foreach ($transactions as $transaction) {
+            $mintEvent = new Event(['sender' => $transaction]);
+
+            Coin::mintCoin($mintEvent);
+        }
```

Comment:
I am wondering if this reflects the current coin status or only the issuing? 

does the current coin holder hold the coins on the smart contract in the end?

Reply:
@bumi sorry i didn't understand your questions , through this code block , i'm doing just `mint()` for already issued coin but which are not present on the smart contracts. 
Should i take into consideration , transfer transactions too ? which means that after the `mint()` action , i should loop on transfer transactions and call POST /transfer ?

Reply:
we have to make sure that the current balances of all users are also correct. 
if we only mint the coins then the balances don't match, do they? 

so we should mint coins for each current holder of those coins. 

//cc @haythem96

Reply:
Ahh yeah, @gdalyy we need to make sure that the balances of the coin holders in the smart contract is the same as in humhub DB, so Instead of minting coins for the issue transaction and then also for the transfer... why not just fetching the coin holders balance and mint for each one of them ? ... what do u think @gdalyy

Reply:
You're right , i'll do the transfer so we make sure that balances on humhub are the same on the blockchain.

Reply:
One thing to mention : i'll be forcing wallet creation for users that do not belong to the space in question and which are holders of the space coin.

Reply:
Done